### DISCUSSION TITLE
Error while datamodule.setup()

### DISCUSSION THREAD

**Discussion by pranaycv** [open-edge-platform/anomalib#2200] [General]
Hi,

I am getting an error while setup of datamodule as below:

ValueError: not enough values to unpack (expected 2, got 0)

I am trying to perform anomaly segmentation task without giving mask, hope that is possible.

Following is the snapshot of the code:

datamodule = Folder(
    name="vials",
    root="E:\PROJECTS\TestAnomalib\TrainSET",
    normal_dir="good",
    abnormal_dir=None,
    task=TaskType.SEGMENTATION,
    mask_dir=None,
    normal_test_dir=None,
    extensions=".bmp",
    normal_split_ratio=0.2,
    image_size=(256, 256),
    train_batch_size=2,
    eval_batch_size=1,
    num_workers=0,
    test_split_mode=TestSplitMode.NONE
)
datamodule.setup()

Your help is greatly appreciated.

**Comment by pranaycv** [comment]
Can anyone help me?

**Comment by samet-akcay** [comment]
You could refer to the following documentation
https://anomalib.readthedocs.io/en/v1.1.0/markdown/guides/how_to/data/custom_data.html#id2

**Reply by pranaycv** [reply]
It works but if we want to implement option #1
1) Train the model and skip the validation and test steps, as we do not have abnormal images to validate and test the model on
What parameters to be set while creating Folder object?

**Reply by samet-akcay** [reply]
# Import the datamodule
from anomalib.data import Folder
from anomalib.data.utils import TestSplitMode

# Create the datamodule
datamodule = Folder(
    name="hazelnut_toy",
    root="datasets/hazelnut_toy",
    normal_dir="good",
    test_split_mode=TestSplitMode.SYNTHETIC,
)

# Setup the datamodule
datamodule.setup()
Option 1 has both normal and abnormal images with masks, as shown below:
```python
# Import the datamodule
from anomalib.data import Folder

# Create the datamodule
datamodule = Folder(
    name="hazelnut_toy",
    root="datasets/hazelnut_toy",
    normal_dir="good",
    abnormal_dir="crack",
    mask_dir="mask/crack",
    normal_split_ratio=0.2,
)

# Setup the datamodule
datamodule.setup()
```

If you don't have masks, and still want to compute the segmentation performance, it is not possible without creating some synthetic anomalies, which is option 2. 

To do that, you need to do the following:

```python
# Import the datamodule
from anomalib.data import Folder
from anomalib.data.utils import TestSplitMode

# Create the datamodule
datamodule = Folder(
    name="hazelnut_toy",
    root="datasets/hazelnut_toy",
    normal_dir="good",
    test_split_mode=TestSplitMode.SYNTHETIC,
)

# Setup the datamodule
datamodule.setup()
```

Anomalib can still perform a segmentation task without using your masks. However, to compute a performance, you do need a ground-truth to compare the performance

**Reply by SPAN-SYSTEMS** [reply]
Hi Samet,

Thanks for your prompt support. You guys have done great job. I was able to train and predict on custom data with ease. The use case was really helpfull:
https://github.com/openvinotoolkit/anomalib/blob/main/notebooks/500_use_cases/501_dobot/501a_training_a_model_with_cubes_from_a_robotic_arm.ipynb
